File: Teoria/drivers_simulator/c/readints.c
// read ints in binary format from System.in and
// write them in text format in System.out

#include <stdio.h>

int main() {
  int i;
  while(fread(&i, sizeof(int), 1, stdin) > 0)
    //we pass a pointer to i
    //reading one-by-one 0 means EOF
    printf("%d\n", i);
}

File: Teoria/drivers_simulator/java/ReadInts.java
import java.io.*;

public class ReadInts {
  public static void main(String[] args) throws IOException {
  // read ints in binary format from System.in and
  // write them in text format in System.out
    DataInputStream in = new DataInputStream(System.in);
    int i;
    while(true){
      try {
        i = in.readInt();
      } catch(EOFException e) {
        break;
      }
      System.out.println(i);
    }
  }
}
